### PR TITLE
Add test to check DNS timeout

### PR DESCRIPTION
**Pull request by aseure**

### CODE REVIEW

**Comment by aseure** [comment]
From #50

File: algoliasearch/client_test.go [hunk 1]
```diff
@@ -976,3 +976,28 @@ func TestSynonyms(t *testing.T) {
...
+		appID + "-2.algolianet.com",
+		appID + "-3.algolianet.com",
+	}
+
+	c := NewClientWithHosts(appID, apiKey, hosts)
+
+	start := time.Now()
+	_, _ = c.ListIndexes()
+
+	if start.Add(15 * time.Second).Before(time.Now()) {
```
Comment (ercolanelli-leo) [inline algoliasearch/client_test.go]: Shouldn't we also set the timeout for the client to 15s?

Comment (aseure) [inline algoliasearch/client_test.go]: It's already the case, isn't it? https://github.com/algolia/algoliasearch-client-go/blob/v2/algoliasearch/transport.go#L46

Comment (ercolanelli-leo) [inline algoliasearch/client_test.go]: Ok works for me :)

**Comment by ercolanelli-leo** [comment]
LGTM :+1:

**Pull request merged**